src/Storages/MergeTree/ReplicatedMergeTreeRestartingThread.cpp

```
@@ -345,7 +351,15 @@ void ReplicatedMergeTreeRestartingThread::partialShutdown(bool part_of_full_shut
 
     storage.partial_shutdown_called = true;
     storage.partial_shutdown_event.set();
-    storage.replica_is_active_node = nullptr;
+
+    try
+    {
+        storage.replica_is_active_node = nullptr;
+    }
+    catch (...)
```

This looks unnecessary since ~EphemeralNodeHolder already catch all exceptions.